code401Challenges/src/main/java/code401Challenges/stacksandqueues/Queue.java

```diff
@@ -0,0 +1,64 @@
+package code401Challenges.stacksandqueues;
+
+public class Queue {
+    private Node front;
+    private Node last;
+    private int count;
+
+    public Queue(){
+        this.front = null;
+        this.last = null;
+        this.count = 0;
+    }
+
+    //getters
+
+    public Node getFront() {
+        return front;
+    }
+
+    public int getCount() {
+        return count;
+    }
+
+
+    //helper functions
+    public boolean isEmpty(){
+        return front == null;
+    }
+
+
+    //feature functions
+
+    public void enqueue(int data){
+        Node n = new Node(data);
+        if(isEmpty() == true){
+            front = n;
+            last = n;
+            count++;
+        } else {
+            last.next = n;
+            last = n;
+            count++;
+        }
+    }
+
+    public int dequeue(){
+        if(isEmpty() == true){
+            System.out.println("Queue is empty");
+        }
+        int value = front.value;
+        front = front.next;
+        count--;
+        return value;
```

Comment:
If the Queue is empty, this will throw an error won't it? It will print out "Queue is empty" and then create `value` which will be `front.next`. If front is null, `front.next` will cause an exception

Reply:
I just verified this, it throws a java.lang.NullPointerException